linking errors with 4.3

I am maintaining the flatpak for QGroundControl (https://flathub.org/apps/org.mavlink.qgroundcontrol). The builds on the CI used to work up until version 4.2.9. With an update to 4.3 (https://github.com/flathub/org.mavlink.qgroundcontrol/pull/9) this fails to link now on the CI and locally.

## Expected Behavior
The package should build successfully, especially given isolated build environments such as flatpak.

## Current Behavior
I am getting "undefined reference to" linking errors like these: https://buildbot.flathub.org/#/builders/37/builds/10326/steps/8/logs/stdio

> `undefined reference to vtable for VehicleGeneratorFactGroup`
> `undefined reference to VehicleGeneratorFactGroup::VehicleGeneratorFactGroup(QObject*)`
> `undefined reference to VehicleEFIFactGroup::VehicleEFIFactGroup(QObject*)`
> ...

## Steps to Reproduce:
Check out the PR branch and run
```sh
app="org.mavlink.qgroundcontrol"
flatpak-builder --ccache --force-clean build-dir ${app}.yml
flatpak-builder --user --install --force-clean build-dir ${app}.yml
```

## System Information
When posting bug reports, include the following information
- Operating System: Ubuntu 22.04
- QGC Version: 4.3.0
- QGC build: flatpak

--- a/src/Vehicle/CMakeLists.txt
+++ b/src/Vehicle/CMakeLists.txt
@@ -73,6 +73,8 @@ add_library(Vehicle
 	VehicleClockFactGroup.h
 	VehicleDistanceSensorFactGroup.cc
 	VehicleDistanceSensorFactGroup.h
+	VehicleEFIFactGroup.cc
+	VehicleEFIFactGroup.h
 	VehicleEscStatusFactGroup.cc
 	VehicleEscStatusFactGroup.h
 	VehicleEstimatorStatusFactGroup.cc
@@ -85,6 +87,8 @@ add_library(Vehicle
 	VehicleGPSFactGroup.h
 	VehicleGPS2FactGroup.cc
 	VehicleGPS2FactGroup.h
+	VehicleGeneratorFactGroup.cc
+	VehicleGeneratorFactGroup.h
 	Vehicle.h
 	VehicleLinkManager.cc
 	VehicleLinkManager.h

--- a/src/Vehicle/CMakeLists.txt
+++ b/src/Vehicle/CMakeLists.txt
@@ -98,8 +98,8 @@ add_library(Vehicle
 	VehicleVibrationFactGroup.h
 	VehicleWindFactGroup.cc
 	VehicleWindFactGroup.h
-        VehicleHygrometerFactGroup.cc
-        VehicleHygrometerFactGroup.h
+	VehicleHygrometerFactGroup.cc
+	VehicleHygrometerFactGroup.h
 	RemoteIDManager.h
 	RemoteIDManager.cc
 

